include/util/bit_range.hpp

```diff
@@ -0,0 +1,95 @@
+#ifndef OSRM_UTIL_BIT_RANGE_HPP
+#define OSRM_UTIL_BIT_RANGE_HPP
+
+#include "util/msb.hpp"
+
+#include <boost/iterator/iterator_facade.hpp>
+#include <boost/range/iterator_range.hpp>
+
+namespace osrm
+{
+namespace util
+{
+
+namespace detail
+{
+template <typename T> std::size_t countOnes(T value)
+{
+    std::size_t number_of_ones = 0;
+    while (value > 0)
+    {
+        auto index = msb(value);
+        value = value & ~(T{1} << index);
+        number_of_ones++;
+    }
+    return number_of_ones;
+}
+
+#if (defined(__clang__) || defined(__GNUC__) || defined(__GNUG__)) && __x86_64__
+inline std::size_t countOnes(std::uint8_t value)
+{
+    return __builtin_popcount(std::uint32_t{value});
+}
+inline std::size_t countOnes(std::uint16_t value)
+{
+    return __builtin_popcount(std::uint32_t{value});
+}
+inline std::size_t countOnes(std::uint32_t value) { return __builtin_popcount(value); }
+inline std::size_t countOnes(std::uint64_t value) { return __builtin_popcountl(value); }
```

Comment:
I think you need `popcountll` (note the `ll` suffix) for 64 bit
- https://gcc.gnu.org/onlinedocs/gcc/Other-Builtins.html
- http://en.cppreference.com/w/cpp/language/types#Properties